test/test_utils/ec2.py

```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
-        f"-e COMMIT_INFO={os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')} "
+        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
         f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} {ecr_uri}"
     )
     try:
-        connection.run(f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
+        connection.run(
+            f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
     except Exception as e:
         raise Exception("Failed to exec benchmark command.\n", e)
     finally:
         connection.run(f"docker rm -f {container_name}")
+    log_location = os.path.join(container_test_local_dir, "benchmark", "logs", log_name)
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, data_source,
+                                                                 threshold, post_process, log_name)
+
+
+def ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location,
+                                                                 data_source, threshold, post_process, log_name):
+    framework = "tensorflow" if "tensorflow" in ecr_uri else "mxnet" if "mxnet" in ecr_uri else "pytorch"
+    framework_version = re.search(r"[0-9]+(\.\d+){2}", ecr_uri).group()
+    py_version = "py2" if "py2" in ecr_uri else "py37" if "py37" in ecr_uri else "py3"
+    processor = "gpu" if "gpu" in ecr_uri else "cpu"
+    work_type = "training" if "training" in ecr_uri else "inference"
+    s3_location = os.path.join(
+        BENCHMARK_RESULTS_S3_BUCKET, framework, framework_version, "ec2", work_type, processor, py_version, log_name
+    )
+    params = {"connection": connection, "log_location": log_location}
+    if "threshold" in signature(post_process).parameters:
+        params["threshold"] = threshold
+    performance_number = post_process(**params)
+    unit = "s" if work_type == "inference" and framework == "tensorflow" \
+        else "ms" if work_type == "inference" and framework == "pytorch" \
+        else "s/epoch" if work_type == "training" and framework == "pytorch" and data_source == "imagenet" \
+        else "images/sec"
+    description = "p99 latency " if unit == "s" or unit == "ms" else ""
+    for k, v in performance_number.items():
+        performance_statement = f'{framework} {framework_version} ec2 {work_type} {processor} {py_version} ' \
+                                f'{data_source} {k} {description}: {v} {unit}, threshold: {threshold[k]} {unit}'
+        connection.run(
+            f"echo {performance_statement} | sudo tee -a {log_location}")
+        LOGGER.info(
+            f"{performance_statement}")
+    connection.run(
+        f"aws s3 cp {log_location} {s3_location}")
+    connection.run(
+        f"echo To retrieve complete benchmark log, check {s3_location} >&2")
```

Comment:
Fix the closing-parenthesis in all these multi-line function calls.

Reply:
Will have them in separate lines